Plugins/SwiftLintPlugin/SwiftLintPlugin.swift

```diff
@@ -3,81 +3,146 @@ import PackagePlugin
+    /// and sets the config file's containing directory as the `BUILD_WORKSPACE_DIRECTORY` value. The package
+    /// directory is used if a config file is not found.
+    ///
+    /// The `BUILD_WORKSPACE_DIRECTORY` environment variable controls SwiftLint's working directory.
+    ///
+    /// Reference: [https://github.com/realm/SwiftLint/blob/0.50.3/Source/swiftlint/Commands/SwiftLint.swift#L7](
+    /// https://github.com/realm/SwiftLint/blob/0.50.3/Source/swiftlint/Commands/SwiftLint.swift#L7
+    /// )
+    private func environment(
+        context: PluginContext,
+        target: Target
+    ) throws -> [String: String] {
+        let workingDirectory: Path = try target.directory.resolveWorkingDirectory(in: context.package.directory)
+        return ["BUILD_WORKSPACE_DIRECTORY": "\(workingDirectory)"]
+    }
+
+    private func makeCommand(
+        executable: PluginContext.Tool,
+        swiftFiles: [Path],
+        environment: [String: String],
+        pluginWorkDirectory path: Path
+    ) -> [Command] {
+        // Don't lint anything if there are no Swift source files in this target
+        guard !swiftFiles.isEmpty
+        else { return [] }
+        // Outputs the environment to the build log for reference.
+        print("Environment:", environment)
+        let arguments: [String] = [
             "lint",
             "--quiet",
             // We always pass all of the Swift source files in the target to the tool,
             // so we need to ensure that any exclusion rules in the configuration are
             // respected.
             "--force-exclude"
         ]
-
         // Determine whether we need to enable cache or not (for Xcode Cloud we don't)
+        let cacheArguments: [String]
         if ProcessInfo.processInfo.environment["CI_XCODE_CLOUD"] == "TRUE" {
-            arguments.append("--no-cache")
+            cacheArguments = ["--no-cache"]
         } else {
-            arguments.append("--cache-path")
-            arguments.append("\(workingDirectory)")
+            let cachePath: Path = path.appending("Cache")
+            do {
+                try FileManager.default.createDirectory(atPath: cachePath.string, withIntermediateDirectories: true)
+            } catch {
+                print(error.localizedDescription)
+            }
+            cacheArguments = ["--cache-path", "\(cachePath)"]
         }
-
-        // Manually look for configuration files, to avoid issues when the plugin does not execute our tool from the
-        // package source directory.
-        if let configuration = packageDirectory.firstConfigurationFileInParentDirectories() {
-            arguments.append(contentsOf: ["--config", "\(configuration.string)"])
+        let outputPath: Path = path.appending("Output")
+        do {
+            try FileManager.default.createDirectory(atPath: outputPath.string, withIntermediateDirectories: true)
```

Comment:
What is this output folder going to be used for?

Reply:
The output directory, although not used by SwiftLint, is a required argument when creating the `.prebuildCommand(...)`.